src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -116,59 +116,82 @@ internal interface ISelfCalibratingPredictor
         IPredictor Calibrate(IChannel ch, IDataView data, ICalibratorTrainer caliTrainer, int maxRows);
     }
 
-    public abstract class CalibratedPredictorBase :
+    /// <summary>
+    /// <see cref="IWeeklyTypedCalibratedPredictor"/> provides a weekly-typed way to access strongly-typed
+    /// <see cref="CalibratedPredictorBase{TSubPredictor, TCalibrator}.SubModelParameters"/> and
+    /// <see cref="CalibratedPredictorBase{TSubPredictor, TCalibrator}.Calibrator"/>.
+    /// <see cref="IWeeklyTypedCalibratedPredictor"/> is commonly used in weekly-typed expressions. The
+    /// existence of this interface is just for supporting existing codebase, so we discourage its uses.
+    /// </summary>
+    [BestFriend]
+    internal interface IWeeklyTypedCalibratedPredictor
```

Comment:
>IWeeklyTypedCalibratedPredictor [](start = 23, length = 31)

I think there is a typo in the naming, should be "weak" if I am not wrong.

Reply:
Ah ha ha... yes somehow I missed this. Funny how the brain just elides over such things as this.

---
In reply to: [254937081](https://github.com/dotnet/machinelearning/pull/2435#discussion_r254937081) [](ancestors = 254937081)

Reply:
Fixed....